flocker/control/httpapi.py

```diff
@@ -134,6 +136,29 @@ def render_if_matches(self, request, **route_arguments):
     return render_if_matches
 
 
```

Comment:
I suggest we move the optimizations in this module to a separate PR and add some explanatory comments justifying  our use of caching.